packages/runtime/src/widgets/Search.tsx

```diff
@@ -52,16 +73,36 @@ export const Search: React.FC<SearchProps> = ({
               }))
           : [],
       );
+    } else if (onValueChange) {
```

Comment:
This should be an else if. The value change handler should fire independently of the other properties. We can't assuming it is only going to be invokeAPI either.